Quantum-Control-Applications/Superconducting/Multiple Flux Tunable Transmons/Use Case 2 - Two Qubit Randomized Benchmarking/README.md

```diff
@@ -0,0 +1,238 @@
+
+```python
+rb = TwoQubitRb(config, bake_phased_xz, {"CZ": bake_cz}, prep, meas, verify_generation=True)
+```
+Before running the experiment, we have to specify the utilized OPX-cluster by creating the *qmm* object with the *QuantumMachinesManager* class. Then, the experiment is executed by calling the run method of the previously generated two-qubit RB program *rb*. Here, we also add important benchmarking parameters like circuit depth (*circuit_depths*), how many different circuits we would like to run per depth (*num_circuits_per_depth*) and how often we we would like to run every circuit (*num_shots_per_circuit*).
+
+```python
+qmm = QuantumMachinesManager('127.0.0.1',8080)
+res = rb.run(qmm, circuit_depths=[1, 2, 3, 4, 5], num_circuits_per_depth=50, num_shots_per_circuit=1000)
+```
+
+The qubit states are measured after the inversion of the random circuit, which ideally is |00>. Due to gate errors, we will also measure the states |01>, |10> and |11>. The result object *res* contains the parameters *circuit_depths*, *num_repeats*, *num_averages* and the result *state*, which is a matrix with values 0,1,2 or 3 corresponding to the possible measurement outcomes. The matrix has the dimension of the given parameters, so for the example code above it will be 5 x 50 x 1000 measured states. To create a figure, containing the histograms for the different circuit dephts we already implemented the function
+
+```python
+res.plot_hist()
+```
+
+To plot the fidelity for increasing circuit depth you can run the function
+
+```python
+res.plot_fidelity()
+```
+
+----------------------------------------------------------------------------------------------------------------------------------------------------------
+# Additional Information
+
+## Implementation in QUA
+The python program *two_qubit_rb_example.py* implements Two-Qubit Randomized Benchmarking with the described setup and the *TwoQubitRb* class. The decomposition of the two-qubit unitaries into CZ and single qubit gates is given in Ref. [^5]. The circuit generation is done using the *baking* tool from the *py-qua-tools* library. Randomization is done prior to the execution using tableau calculation, also to find the inverse operation. The sequences are passed to the OPX using *input stream*.
```

Comment:
randomization is done using a PRNG, the inverse calculation is done using Clifford tableaus